src/libsystemd/sd-journal/journal-file.c

```diff
@@ -3071,6 +3081,16 @@ static int generic_array_bisect(
 
                         for (;;) {
                                 if (left == right) {
+                                        if (m != m_original && direction == DIRECTION_DOWN) {
+                                                /* We found corrupted entries, and the final index may not be tested yet. */
+                                                r = generic_array_bisect_step(f, array, left, needle, test_object, direction, &m, &left, &right);
+                                                if (r < 0)
+                                                        return r;
+                                                if (IN_SET(r, TEST_GOTO_PREVIOUS, TEST_GOTO_NEXT))
+                                                        return 0; /* The entry does not pass the test. */
+                                                assert(left == right);
```

Comment:
Should we assert `r == TEST_FOUND` here?

Reply:
No, `generic_array_bisect_step()` never returns `TEST_FOUND`.

Reply:
Actually, `r == TEST_RIGHT`, in this case.